### ISSUE TITLE
Trouble displaying Results from dynamic source

### ISSUE THREAD

**Issue by TheKingUnderTheMountain** [running-coder/jquery-form-validation#27]
Hello,

First off, really great website and documentation. Keep up the good work!
I'm trying to get some location results from a webservice, and while I get data back, I can't seem to get it to show up. I just get the Empty Template.

Here's my TypeAhead:
```
$(document).ready(function () {
        $.typeahead({
            input: ".js-typeahead",
            minLength: 1,
            order: "asc",
            dynamic: true,
            delay: 500,
            display: "PlaceId",
            emptyTemplate: "no result for {{query}}",
            template: function (query, item) {
                return '<span data-placeid="{{PlaceId|raw}}">{{PlaceId}} - <small>{{PlaceName}</small></span>'
            },
            source: {
                
                    // Ajax Request
                    ajax: function (query) {
                        return {
                            type: "POST",
                            url: "/Home/LocationJson",
                            path: "data",
                            data: {
                                q: "{{query}}"
                            },
                            callback: {
                                done: function (data) {
                                    console.log(data);
                                    return data;
                                }
                            }
                        }
                    }
    
            },
            callback: {
                onInit: function (node) {
                    console.log('Typeahead Initiated on ' + node.selector);
                }
            },
            debug: true
        });
    });
```

Here's a screenshot of the XHR and the object returned.
http://screencast.com/t/fL7LDk7c

I've made a number of adjustments to the data being returned trying to get rid of the debug errors, and was able to get rid of a couple of them. But it still isn't returning data, so I'm not sure what's going on there. 

Here's the HTML on the page:
```
<form>
    <div class="typeahead__container">
        <div class="typeahead__field">

            <span class="typeahead__query">
                <input class="js-typeahead"
                       name="q"
                       type="search"
                       autocomplete="off">
            </span>
            <span class="typeahead__button">
                <button type="submit">
                    <span class="typeahead__search-icon"></span>
                </button>
            </span>

        </div>
    </div>
</form>
```

Thanks for taking a look - I appreciate it!

**Comment by running-coder** [comment]
Hey, perhaps you have created the ticket under the wrong project ;) np
1. is your data already filtered by your API? 
    - in that case, you might want to specify `maxItem: 0` to display all your results
    - also, use the option `filter: false` on your source group to ignore Typeahead filtering (since you already filtered in BE
2. What does it print in your console.log(data) of the callback.done function?

**Comment by TheKingUnderTheMountain** [comment]
Oh, shoot. You're absolutely right that this is the wrong support forum. I had been browsing the correct forum for a while - I wonder how I ended up here. Sorry!

I'll try out your thoughts on #1, as the data is filtered already when it comes to me.

As for #2, I've attached a screenshot for you.
![2016-05-26_1553](https://cloud.githubusercontent.com/assets/7153099/15588105/5de23796-235a-11e6-9337-c072f1bd8e7a.png)

Thanks for taking the time!

**Comment by running-coder** [comment]
Yea I can see "gran" does not match `PlaceId` from your data[0] item so `filter:false` would probably resolve your issue. Anything suspicious inside the Typeahead debug messages?

**Comment by TheKingUnderTheMountain** [comment]
I don't know about you, but I often fix computer issues for friends or family where they have some crazy problem, I show up to look at it, and it magically fixes itself. They're left rambling on about how they tried all those things and it really wasn't working before I got there.

Seems this must happen to you, too. I have a feeling I did fix it when I was working through the debug messages, but it wanted to pretend it was still broken. Posting this message and getting your reply back appears to have fixed it, as I haven't changed anything from where it last was and now it works fine.

Sorry to bother you! Thanks for fixing it by taking a look :)
